### ISSUE TITLE
Add av cherry-pick

### ISSUE THREAD

**Issue by draftcode** [aviator-co/av#248]

**Comment by jainankit** [comment]
> is this issue open for contributions?
Yes, absolutely. @draftcode can you add some more context to this ticket?

**Comment by kolharsam** [comment]
Hey @draftcode @jainankit 
Just wanted to provide a gentle nudge regarding this!

Thanks!

**Comment by jainankit** [comment]
Let me add the context. `git cherry-pick` as a command already exist: https://git-scm.com/docs/git-cherry-pick

When working with av CLI, it would be nice to have a more native experience of managing the cherry-pick. So in addition to the typical cherry-pick behavior where a commit is added to the current branch, this command should also run the `av stack restack` command internally, so that the entire stack is updated.

Note that this command should NOT be pushing the changes to remote.

It should support all capabilities of the standard `git cherry-pick`, and should gracefully handle scenarios where the cherry-pick fails due to conflict, allowing user to continue after resolving the issue.

```
av cherry-pick [--edit] [-n] [-m <parent-number>] [-s] [-x] [--ff]
		  [-S[<keyid>]] <commit>…​
av cherry-pick (--continue | --skip | --abort | --quit)
```

If the additional params (`[--edit] [-n] [-m <parent-number>] [-s] [-x] [--ff][-S[<keyid>]]`) are too complicated to support, it's ok to not support them at the beginning, but we should still handle the failure scenarios and not leave dev env in a bad state.

**Comment by kolharsam** [comment]
Hey folks! @jainankit @draftcode 

Sorry that I haven't been able to work on the issue as of yet! I'm just busy with all of the tasks during this part of the semester. I will give you folks an update in a couple of weeks time.

Appreciate your understanding!

**Comment by kolharsam** [comment]
Hey @draftcode @jainankit 

I've started work on this https://github.com/aviator-co/av/pull/495 sorry I took a long time to get back to this, but yeah, the semester ended yesterday and I'm here on it.

I would appreciate any feedback on the current status, but yeah, thanks again for the opportunity to contribute.
